src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -81,18 +81,11 @@ impl FileType {
 impl TargetInfo {
     pub fn new(
         config: &Config,
-        requested_target: &Option<String>,
+        requested_kind: CompileKind,
```

Comment:
The [rationale](https://github.com/rust-lang/cargo/blob/494cbd8c8d066572fcee6e41ffa3aca8ee316104/src/cargo/core/compiler/build_context/target_info.rs#L390-L408) is basically that `RUSTFLAGS` plus `--target` shouldn't pass the flags to build scripts. This is needed here I think because we calculate `TargetInfo` for both the host and target, right?

Reply:
Ah, I see now.